lib.rs

```diff
@@ -65,17 +116,19 @@ impl<D: Decorator> Format<D> {
         }
     }
 
+    // Returns `true` if message is empty
     fn print_msg_header(&self,
                         io: &mut io::Write,
                         rd: &D::RecordDecorator,
                         record: &Record)
-                        -> io::Result<()> {
+                        -> io::Result<bool> {
         try!(rd.fmt_timestamp(io, &*self.fn_timestamp));
         try!(rd.fmt_level(io,
                           &|io: &mut io::Write| write!(io, " {} ", record.level().as_short_str())));
 
-        try!(rd.fmt_msg(io, &|io| write!(io, "{}", record.msg())));
-        Ok(())
+        let mut writer = CountingWriter::new(io, true);
+        try!(rd.fmt_msg(&mut writer, &|io| write!(io, "{}", record.msg())));
```

Comment:
Don't you just want to use `CountingWriter` inside the `&|io| write!(io, "{}", record.msg())` closure and write the count (or `count == 0`) to variable borrowed from the `print_msg_header` scope? This way you don't have to worry about control sequences introduced by `fmt_msg` since you don't have to count them at all.

If so, it seems to me that it would be a good idea to not print control sequences at all in `fmt_msg` if there was no output.